### PR TITLE
Where is anchor applied 

### PR DESCRIPTION
**Pull request by oluoluoxenfree**
## Steps to test/reproduce
_Please explain how to best reproduce the issue and/or test the changes locally (including the pages/URLs/views/states to review)._

## Show me
_Provide screenshots/animated gifs/videos if necessary._

# REMEMBER: Attach this PR to the Trello card

### CODE REVIEW

**Comment by oluoluoxenfree** [comment]
[I want to always know the property where an `anchor()` fn is being applied.](https://trello.com/c/Ngg4pSm9/56-i-want-to-always-know-the-property-where-an-anchor-fn-is-being-applied)

File: src/parse.ts [hunk 1]
```diff
@@ -290,6 +303,28 @@ export function parseCSS(css: string) {
...
+        const anchorFnData = customProperties[name];
+        if (anchorFnData) {
+          anchorFunctions[rule.value] = {
+            ...anchorFunctions[rule.value],
+            [this.declaration.property]: anchorFnData,
+          };
+        }
+      }
+    });
+  }
```
Comment (jgerigmeyer) [inline src/parse.ts:326]: @oluoluoxenfree The goal of the second pass is to find anywhere a custom property is used (e.g. `--center`), and store the inset property it's being used by (e.g. `top`).

I started playing with this just to check if my basic understanding was correct, and I was surprised but I think it actually already works okay as-is 😲 -- but we should test it, and I'm sure there's more cleanup we could do here. Take a look and see what you think.

File: src/parse.ts [hunk 2]
```diff
@@ -171,13 +177,20 @@ function getAnchorNameData(node: csstree.CssNode, rule?: csstree.Raw) {
...
   rule?: csstree.Raw,
 ) {
   if (isAnchorFunction(node) && rule?.value && declaration) {
-    return { [declaration.property]: parseAnchorFn(node) };
+    const data = parseAnchorFn(node);
+    if (declaration.property.startsWith('--')) {
+      customProperties[declaration.property] = data;
+      return;
+    }
+    return { [declaration.property]: data };
```
Comment (jgerigmeyer) [inline src/parse.ts:193]: We should probably check here to see if `declaration.property` is a valid inset property (top/right/bottom/left).

File: src/parse.ts [hunk 3]
```diff
@@ -290,6 +303,28 @@ export function parseCSS(css: string) {
...
 
+  // Find where CSS custom properties are used
+  if (Object.values(customProperties).length > 0) {
+    csstree.walk(ast, function (node) {
+      const rule = this.rule?.prelude as csstree.Raw | undefined;
+      if (
+        rule?.value &&
+        isVarFunction(node) &&
+        node.children.first &&
+        this.declaration
```
Comment (jgerigmeyer) [inline src/parse.ts:314]: We should probably also check here to see if `this.declaration.property` is a valid inset property (top/right/bottom/left).

File: src/parse.ts (hunk 2 above)
Comment (oluoluoxenfree) [reply src/parse.ts:193]: @jgerigmeyer what should we do if it isn't? just return?

File: src/parse.ts [hunk 4]
```diff
@@ -171,13 +177,24 @@ function getAnchorNameData(node: csstree.CssNode, rule?: csstree.Raw) {
...
+    const data = parseAnchorFn(node);
+    if (declaration.property.startsWith('--')) {
+      customProperties[declaration.property] = data;
+      return;
+    }
+    if (isInset(declaration.property)) {
+      return { [declaration.property]: data };
+    } else {
+      return;
+    }
```
Comment (jgerigmeyer) [inline src/parse.ts:197]:
```suggestion
    }
```

File: src/parse.ts [hunk 5]
```diff
@@ -193,6 +210,16 @@ function getPositionFallbackDeclaration(
...
 }
 
+function isInset(property: string) {
+  const insetProperties: string[] = ['left', 'right', 'top', 'bottom'];
+
+  if (insetProperties.includes(property)) {
+    return true;
+  } else {
+    return false;
+  }
```
Comment (jgerigmeyer) [inline src/parse.ts:220]:
```suggestion
  const insetProperties = ['left', 'right', 'top', 'bottom'];

  return insetProperties.includes(property);
```

**Review by jgerigmeyer — APPROVED** [review]

**Pull request merged**